src/identity/components/userprofile/UserDefaults.tsx

```diff
@@ -58,8 +59,10 @@ export const UserDefaults: FC = () => {
 
   const defaultAccount = useMemo(
     () =>
-      accounts ? accounts.find(el => el.isDefault === true) : emptyAccount,
-    [accounts]
+      accounts
+        ? accounts.find(acct => acct.id === defaultAccountId)
+        : emptyAccount,
+    [accounts, defaultAccountId]
   )
```

Comment:
Before, this component didn't need to check for updates in the `defaultAccountId` loaded from context because the context forced a new api call to `/accounts`. Now, no call to `/accounts` is needed here either - we can simply retrieve the accounts array from context when the default account id changes.